cpp/ql/lib/experimental/semmle/code/cpp/semantic/analysis/FloatDelta.qll

```diff
@@ -16,6 +16,5 @@ module FloatDelta implements DeltaSig {
   Delta fromInt(int n) { result = n }
 
   bindingset[f]
-  bindingset[result]
-  Delta fromFloat(float f) { result = f }
+  Delta fromFloat(float f) { result = min(float diff, float res | diff = (res - f).abs() and res = f.ceil() or diff = (f - res).abs() and res = f.floor() | res order by diff )}
```

Comment:
Are the calls to `abs` really needed? When `res = f.ceil()` I'd expect `res >= f`, and so `(res - f).abs()` should hopefully be equal to `res - f`?

(And dually for the other case.)

Reply:
You're right, they aren't.